### ISSUE TITLE
Implement dashboard endpoint and lecturer authentication

### ISSUE THREAD

**Issue by nickykay98** [UoaWDCC/VPS#29]
## Issue Description

Create a page that’s only accessible to lecturers when they log in where they can see the dashboard (and other students can't access it when they log in)

See minutes from [3/8/22](https://docs.google.com/document/d/1tK8Goe7VbGD-C0zOAQzgCRt4Zg3RCA6aZ1Z6QVdvDkQ/edit#) and [27/7/22](https://docs.google.com/document/d/12zVyaj8VlxgzpDsclpp5i6_1j1ORm3zQVtTEurwiEaI/edit#) for more details

## User Story

"As a lecturer, I want to be able to login and access my dashboard from the home page"

"As a lecturer, I want to be able to login and see the option to access my dashboard without letting other students access it" 

## MVP

A button only visible to lecturers to access their dashboard (which could just be a blank page for now)

## Tasks
 
- [x] Create a page and endpoint within the app for where the dashboard will eventually be accessed - Matthew
- [x] Create a button on the home page linking to the page above and only shows for lecturers - Matthew
- [x] Get people to log into their Google account before playing a scenario via the shareable link - Lia
- [x] Show the dashboard button only if the authenticated user is a lecturer - Lia
- [x] Bring the authentication to the backend to improve security (see [Raymond's diagram](https://docs.google.com/document/d/1tK8Goe7VbGD-C0zOAQzgCRt4Zg3RCA6aZ1Z6QVdvDkQ/edit#bookmark=id.vov2jxzgbxm) for more info) - Lia

**Comment by nickykay98** [comment]
@lia-arroyo @mlai962 feel free to add any other details that I might have missed and don't hesitate to reach out to me/@Harshal-D and contact each other! 😊

**Comment by nickykay98** [comment]
@lia-arroyo Just an idea from the other issue that you could explore for lecturer authentication! https://github.com/UoaWDCC/VPS/issues/30#issuecomment-1186822161

**Comment by lia-arroyo** [comment]
Current progress with authentication:
Commits can be found at branch: [VPS-029/authentication-and-dashboard-button](https://github.com/UoaWDCC/VPS/tree/VPS-029/authentication-and-dashboard-button)
- Matthew has created a button that leads to an empty dashboard page. He tried to reuse the top bar that they already created but there were errors.
- Lia has coded a hacky way of determining whether or not the current user is a lecturer, depending on the email domain. However, security is an issue as it is all done on the frontend.

Next steps:
- Lia to bring the authentication to the backend to improve security.

**Comment by nickykay98** [comment]
@lia-arroyo Me and @Harshal-D have decided to keep this issue open for sprint 4 so that you can keep working on the authentication stuff - I've added details about this from the sprint meeting at the top

@mlai962 We'll have you working on the dashboard implementation with Alfred, keep an eye out for the new story! 

In the meantime, could you please create a PR to merge any changes made on this branch into main? Maybe @mlai962 can do this so that your frontend work is accessible to everyone - @lia-arroyo you might just want to make sure your existing work that would be merged into main as well isn't breaking anything 😅

**Comment by nickykay98** [comment]
Reopening for Lia to work on for sprint 4

**Comment by nickykay98** [comment]
Just popping this diagram from Ray shared on Slack during today's dev session
<img width="968" alt="image" src="https://user-images.githubusercontent.com/49134354/183811217-78ac1e6d-9e0e-4d5b-8419-3c81bdb69485.png">

**Comment by nickykay98** [comment]
Continuing to be worked on in sprint 5

**Comment by lia-arroyo** [comment]
Update on [PR to Main](https://github.com/UoaWDCC/VPS/pull/42)

**Comment by nickykay98** [comment]
Continuing to be worked on in sprint 7

**Comment by lia-arroyo** [comment]
Implemented lecturer authentication in #42. Just need to update the dashboard button once #46 is merged to main, and this issue should be finished. :)

**Comment by nickykay98** [comment]
@lia-arroyo #46 is merged :)

**Comment by lia-arroyo** [comment]
Thanks @nickykay98 :)

I'm just finishing up some assignments this weekend, but I'll finish up the dashboard button tomorrow or Tuesday.

**Comment by lia-arroyo** [comment]
Hey @nickykay98 ,

After pulling from the master branch, I can't get the app to run properly. I think this is because of what @mlai962 said in #32. This screenshot is after running `yarn install`:

![image](https://user-images.githubusercontent.com/94775011/197513157-85094b2e-7362-454e-a825-4a51d6423f5f.png)
![image](https://user-images.githubusercontent.com/94775011/197513233-0947cf8c-4671-49e1-866c-755e522a4ea1.png)

And this is what I get after `yarn run init` :
![image](https://user-images.githubusercontent.com/94775011/197514149-1cca0838-a8ef-49b2-be24-1c26143994e2.png)
![image](https://user-images.githubusercontent.com/94775011/197514179-3a47f4bf-f192-4c63-beb1-d8b02517fe76.png)

I also don't know how to resolve these. 

The dashboard button in the sidebar of the Scenario Selection page already checks for whether the authenticated user is staff or not though, so my task was just going to be moving this button to the Scene Selection page (if that is what we still prefer), and checking that it works. This should be a really easy once this has been resolved :)

**Comment by mlai962** [comment]
I'm fairly sure I got the same errors after Alfred first implemented the scene graph, we didn't know how to solve them, but recloning the repo got rid of the errors.

**Comment by nickykay98** [comment]
> I'm fairly sure I got the same errors after Alfred first implemented the scene graph, we didn't know how to solve them, but recloning the repo got rid of the errors.

@lia-arroyo 👆
